### PR TITLE
DevWs for CoolKit Technology Co.,Ltd containing containing Ikea Button

### PR DESCRIPTION
**Pull request by mingwei0827**

### CODE REVIEW

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 1]
```diff
@@ -374,6 +384,20 @@ private Map getButtonEvent(Map descMap) {
...
+			buttonNumber = 1
+			if (descMap.commandInt == 0x00) {
+				buttonState = "held"
+			} 
+			else if (descMap.commandInt == 0x01) {
+				buttonState = "double"
+			}
+			else {
+				buttonState = "pushed"
+			}
```
Comment (KKlimczukS) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: Please, could You reformat this section ?
`} else if {`

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy (hunk 1 above)
Comment (mingwei0827) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: I have modified the format, please recheck.

**Comment by mingwei0827** [comment]
@KKlimczukS  Please help me recheck the code for other problems, thank you.

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 2]
```diff
@@ -33,6 +33,8 @@ metadata {
 		fingerprint manufacturer: "KE", model: "TRADFRI open/close remote", deviceJoinName: "IKEA Remote Control", mnmn: "SmartThings", vid: "SmartThings-smartthings-IKEA_TRADFRI_open/close_remote" // raw description 01 0104 0203 01 07 0000 0001 0003 0009 0020 1000 FC7C 07 0003 0004 0006 0008 0019 0102 1000 //IKEA TRÅDFRI Open/Close Remote
 		fingerprint manufacturer: "SOMFY", model: "Situo 4 Zigbee", deviceJoinName: "SOMFY Remote Control", mnmn: "SmartThings", vid: "SmartThings-smartthings-Somfy_Situo4_open/close_remote" // raw description 01 0104 0203 00 02 0000 0003 04 0003 0005 0006 0102
 		fingerprint manufacturer: "SOMFY", model: "Situo 1 Zigbee", deviceJoinName: "SOMFY Remote Control", mnmn: "SmartThings", vid: "SmartThings-smartthings-Somfy_open/close_remote" // raw description 01 0104 0203 00 02 0000 0003 04 0003 0005 0006 0102
+		fingerprint inClusters: "0000, 0001, 0003",  outClusters: "0003, 0006", manufacturer: "eWeLink", model: "WB01", deviceJoinName: "eWeLink Button" //eWeLink Button
+		fingerprint inClusters: "0000, 0001, 0003, 0020, FC57",  outClusters: "0003, 0006, 0019", manufacturer: "eWeLink", model: "SNZB-01P", deviceJoinName: "eWeLink Button" //eWeLink Button
```
Comment (KKlimczukS) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: single space is enough before `outClusters`

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy (hunk 2 above)
Comment (mingwei0827) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: @KKlimczukS I've reformatted it~

**Review by KKlimczukS — APPROVED** [review]

**Comment by mingwei0827** [comment]
@KKlimczukS  hi, I want to ask how long it will take for this to be published, and the "publish Request" is still displayed in DevWs, thank you.
lQLPDhrxhFr9YRvNAyvNBOawuzx33bA-RP0BuPG7kQA4AA_1254_811

**Comment by KKlimczukS** [comment]
@mingwei0827 Please report it here: 
https://support.smartthings.com/hc/en-us -> `Email`

**Comment by k-morawski** [comment]
Dear @mingwei0827 , please let me add that I checked DE catalog and your device is visible there. I believe it is also visible in all countries you selected. The problem is probably connected with our systems synchronization.

**Comment by mingwei0827** [comment]
@k-morawski Hello, I'd like to ask, in this code request merge, two devices ("SNZB-01P" and "WB01") are included.
However, "WB01" in DevWs, I did not go to the next step (request to push), so when the code has been merged, do I need to continue to the next step?
thank you~
image
image

**Comment by k-morawski** [comment]
Thank you @mingwei0827 , I sent you an email with the problem description. Could you please reply to it and let me know if I understand it correctly. Moreover I asked for DevWS team for help.

**Pull request merged**